script.js

```diff
@@ -1,4 +1,350 @@
+// The player who is closer to, but not above 21 wins the hand.
+
+// Base Version
+
+// Global variables i.e. Game modes
+var GAME_START = "game start";
+var GAME_CARDS_DRAWN = "cards are drawn";
+var GAME_RESULTS_SHOWN = "results are shown";
+var GAME_HIT_OR_STAND = "hit or stand";
+var currentGameMode = GAME_START;
+
+// Define hands
+var playerHand = [];
+var dealerHand = [];
+
+// Define variable to hold deck of cards
+var gameDeck = [];
+
+//Helper Functions
+
+// Function that creates a deck of cards, used by createNewDeck function
+var createDeck = function () {
+  // deck array
+  var deck = [];
+  // for 'while loop' to create suits for cards
+  var suits = ["Diamonds", "Clubs", "Hearts", "Spades"];
+  var indexSuits = 0;
+  while (indexSuits < suits.length) {
+    var currSuit = suits[indexSuits];
+    // 13 ranks... ace to king - rank to define "card positions"
+    var indexRanks = 1;
+    while (indexRanks <= 13) {
+      var cardName = indexRanks;
+      // define card value - differentiate from rank: 'ace' = 1 / 11, 'jack' & 'queen' & 'king' = 10
+      if (cardName == 1) {
+        cardName = "ace";
+        // define ace value as 11 all the way. if handValue > 10, -11 to total value
+        // vs. coding a function to redefine the value for ace
+      }
+      if (cardName == 11) {
+        cardName = "jack";
+      }
+      if (cardName == 12) {
+        cardName = "queen";
+      }
+      if (cardName == 13) {
+        cardName = "king";
+      }
+      var card = {
+        name: cardName,
+        suit: currSuit,
+        rank: indexRanks,
+      };
+      deck.push(card);
+      indexRanks = indexRanks + 1;
+    }
+    indexSuits = indexSuits + 1;
+  }
+  return deck;
+};
+// Function that generates a random number, used by shuffle deck function
+var getRandomIndex = function (size) {
+  return Math.floor(Math.random() * size);
+};
+
+// Function that shuffles a deck, used by createNewDeck function
+var shuffleDeck = function (cards) {
+  var index = 0;
+  while (index < cards.length) {
+    var randomIndex = getRandomIndex(cards.length);
+    var currentItem = cards[index];
+    var randomItem = cards[randomIndex];
+    cards[index] = randomItem;
+    cards[randomIndex] = currentItem;
+    index = index + 1;
+  }
+  return cards;
+};
+
+// Function that creates and shuffles a deck
+var createNewDeck = function () {
+  var newDeck = createDeck();
+  var shuffledDeck = shuffleDeck(newDeck);
+  return shuffledDeck;
+};
+
+// Function that checks a hand for black jack
+var checkForBlackJack = function (handArray) {
+  // Loop through player hand
+  // if there is a blackjack return true
+  // else return false
+  var playerCardOne = handArray[0];
+  var playerCardTwo = handArray[1];
+  var isBlackJack = false;
+
+  // Possible black jack scenerios
+  // First card is Ace +  Second card is 10 or suits
+  // Second card is Ace +  First card is 10 or suits
+  if (
```

Comment:
You almost never need/want to store a true/false directly in a variable. In this case, you could have done it as follows:
```js
return (playerCardOne.name == "ace" && playerCardTwo.rank >= 10) ||     (playerCardTwo.name == "ace" && playerCardOne.rank >= 10)
```